test/CadlRanchProjects.Tests/encode-bytes.cs

```diff
@@ -161,5 +161,42 @@ public class EncodeBytesTests : CadlRanchTestBase
             Response response = await new BytesClient(host, null).GetRequestBodyClient().Base64urlAsync(data);
             Assert.AreEqual(204, response.Status);
         });
+
+        [Test]
+        public Task Encode_Bytes_ResponseBody_default() => Test(async (host) =>
+        {
+            var response = await new BytesClient(host, null).GetResponseBodyClient().DefaultAsync();
+            BinaryData.Equals(BinaryData.FromObjectAsJson("dGVzdA=="), response.Value);
```

Comment:
as far as I know, `BinaryData` class does not have a static method `BinaryData.Equals(BinaryData, BinaryData)` defined, therefore here we are actually calling `object.Equals`, which is fine, but since we are in a test case, we need to assert that these two BinaryData instances are the same, we need to assert the result. This `Equals` method just returns true or false, it does not assert.

We should call either:
```csharp
Assert.AreEqual(BinaryData.FromObjectAsJson("dGVzdA=="), response.Value);
```
I believe this method will call the `Equals` method on the instance to compare the instances instead of just checking reference equal, or you could do this:
```csharp
Assert.IsTrue(BinaryData.Equals(BinaryData.FromObjectAsJson("dGVzdA=="), response.Value));
```
I prefer the first if it works because it has more semantics.

Reply:
fixed